### PR TITLE
Typings: FindOrCreateOptions uses FindOptions and CreateOptions

### PR DESCRIPTION
**Pull request by AuHau**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

FindOrCreateOptions uses FindOptions and CreateOptions

### CODE REVIEW

**Comment by AuHau** [comment]
Related to https://github.com/sequelize/sequelize/issues/11560

File: types/lib/model.d.ts [hunk 1]
```diff
@@ -684,17 +684,7 @@ export interface Hookable {
...
-   * A hash of search attributes.
-   */
-  where: WhereOptions;
-
-  /**
-   * Default values to use if building a new instance
-   */
-  defaults?: object;
-}
+export type FindOrCreateOptions = FindOptions & CreateOptions
```
Comment (sushantdhiman) [inline types/lib/model.d.ts]: You still want to specify `defaults` option which is not covered by either `FindOptions` or `CreateOptions`

```js
types/test/model.ts:96:3 - error TS2345: Argument of type '{ where: { username: string; }; defaults: { beta_user: boolean; }; }' is not assignable to parameter of type 'FindOrCreateOptions'.

  Object literal may only specify known properties, and 'defaults' does not exist in type 'FindOrCreateOptions'.

96   defaults: {

     ~~~~~~~~~~~

97     beta_user: true

   ~~~~~~~~~~~~~~~~~~~

98   }

   ~~~
```

**Comment by sushantdhiman** [comment]
Closing this PR, please comment or open a new fixed PR if you want to continue

**Comment by jurajpiar** [comment]
Could we re-open this issue?
I've changed it to:
```
/**
 * Options for Model.findOrCreate method
 */
export type FindOrCreateOptions = FindOptions & CreateOptions & {
  /**	
   * Default values to use if building a new instance	
   */	
  defaults?: object;
}
```

I was unable to push the branch due to permission though. ❓

**Pull request closed without merging**